layers/+spacemacs/spacemacs-defaults/packages.el

```diff
@@ -75,7 +75,13 @@
     "ad" 'spacemacs/dired
     "fj" 'dired-jump
     "jd" 'dired-jump
-    "jD" 'dired-jump-other-window))
+    "jD" 'dired-jump-other-window)
+  (when (eq 'vim dotspacemacs-editing-style)
+    (evil-define-key 'normal dired-mode-map (kbd "n") 'evil-ex-search-next)
+    (evil-define-key 'normal dired-mode-map (kbd "N") 'evil-ex-search-previous))
+  (when (eq 'hybrid dotspacemacs-editing-style)
+    (evil-define-key 'normal dired-mode-map (kbd "n") 'evil-search-next)
+    (evil-define-key 'normal dired-mode-map (kbd "N") 'evil-search-previous)))
```

Comment:
How about something like following?
```elisp
(when (memq dotspacemacs-editing-style '(hybrid vim))
    (evil-define-key 'normal dired-mode-map (kbd "n") 'evil-ex-search-next)
    (evil-define-key 'normal dired-mode-map (kbd "N") 'evil-ex-search-previous))
```

Reply:
I should have specified the reason for the different commands.

The `evil-search-module` values are different for both editing styles:
- `vim` = `evil-search`
- `hybrid` = `isearch`

With `isearch` the command `evil-ex-search-next` returns:
>Search failed: "No search pattern"

And it doesn't work the other way around, `evil-search-next` with the `vim` editing style returns:
>"'vim": pattern not found

(`'vim` was the last search pattern when the `evil-search-module` value was `isearch` (hybrid))

I'll add a comment about the `evil-search-module` difference above the key bindings, 
and to the commit message, to make it clear why they use different commands.

Reply:
I had missed the reasoning. This clarifies.